libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/VoiceInstructionsPredownloadHub.kt

```diff
@@ -0,0 +1,47 @@
+package com.mapbox.navigation.ui.voice.api
+
+import androidx.annotation.VisibleForTesting
+import com.mapbox.navigation.core.internal.LegacyMapboxNavigationInstanceHolder
+import com.mapbox.navigation.core.internal.MapboxNavigationCreateObserver
+
+private data class TriggerAndObserver(
+    val trigger: VoiceInstructionsDownloadTrigger,
+    val observer: MapboxNavigationCreateObserver,
+)
+
+internal object VoiceInstructionsPredownloadHub {
+
+    private val registrants = hashMapOf<MapboxSpeechLoader, TriggerAndObserver>()
+
+    fun register(loader: MapboxSpeechLoader) {
+        if (loader !in registrants) {
+            val trigger = VoiceInstructionsDownloadTrigger(
+                3 * 60,
+                0.5
```

Comment:
Where are these magic numbers coming from? 🤔

Reply:
These are just the defaults. Switched to using named constants.